scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/util/DataValidationHelper.java

```diff
@@ -168,6 +157,52 @@ public static boolean isElementUnique(int[] array) {
 		return overlapDateTimeValidationErrorMsgList;
 	}
 
+	private static boolean isStartEndDateOverlapping(RecurringScheduleTime current, RecurringScheduleTime next) {
+		boolean isOverlapping;
+
+		// NOTE: The Start and End Dates in the schedules are not sorted, so we need to compare dates
```

Comment:
We had thought about sorting the dates.Sorting current and next date is a bit complex since these dates are optional and can have same start date and different end dates and vice versa...plus many more combinations. We will try again to find some way out.